### ISSUE TITLE
Decide whether to use TTF of OTF fonts

### ISSUE THREAD

**Issue by dalboris** [vgc/vgc#173]
Source Sans Pro and Source Code Pro are available either as *.ttf files (TrueType) or *.otf files (OpenType). See:

https://github.com/adobe-fonts/source-code-pro
https://github.com/adobe-fonts/source-sans-pro

I believe using OpenType is a better choice. It is a newer format, using cubic Bezier instead of quadratic, and the Source Sans Pro and Source Code Pro fonts are designed for OpenType, as we can read in the [master branch of the project](https://github.com/adobe-fonts/source-code-pro/tree/master):

> The key to building the OTF fonts is makeotf, which is part of the AFDKO toolset. Information and usage instructions can be found by executing makeotf -h. The TTFs are generated with the otf2ttf and ttfcomponentizer tools.

However, in VGC, we are currently using the TrueType files. I don't remember if this was a deliberate choice (e.g., supporting older operating systems), or if is was a mistake. I should try out the OpenType fonts instead, and keep them if they seem supported on all relevant platforms.

Note: italic versions of Source Code Pro are now available (they were not yet available when I first added the fonts to VGC). We should add them.

**Comment by dalboris** [comment]
Let's try to use the OTF fonts instead of the TTF fonts. I'm downloading the fonts from the GitHub repositories:

https://github.com/adobe-fonts/source-sans-pro/releases
https://github.com/adobe-fonts/source-code-pro/releases

However, note that the two latest GitHub releases of Source Code Pro (Aug 18, 2017 and Feb 11, 2018) are a completely new version using "variable fonts" technology, see:

https://blog.typekit.com/2016/09/14/variable-fonts-a-new-kind-of-font-for-flexible-design/

This is honestly pretty cool, but as noted in the release note, not well supported at all yet:

> Warning: These .otf fonts CANNOT be installed on Windows or macOS systems because the OSes do not currently support the OpenType-CFF2 format. Install the latest non-variable fonts instead.

Therefore, we don't want to use them for VGC, and instead use the latest non-variable version, which is from Jul 19, 2016.

In the case of Source Sans Pro, we should be able to use the latest version 3.006 from Sep 5, 2019.

**Comment by dalboris** [comment]
Here are some results after doing a few tests.

On Linux, the OFT font seemed to look similar as the TTF font.

On Windows 10, on a 1920x1200 27" LCD display, the OTF fonts look significantly different (worse, in my opinion) than the TTF fonts

Using the TTF fonts:

![image](https://user-images.githubusercontent.com/4809739/65375142-6f2bfb00-dc92-11e9-8b5a-4f7a34cb3e28.png)
![image](https://user-images.githubusercontent.com/4809739/65375179-ccc04780-dc92-11e9-85e2-2df309c58e2f.png)

Using the OTF fonts:

![image](https://user-images.githubusercontent.com/4809739/65375194-ff6a4000-dc92-11e9-9f5b-95879ce1313a.png)
![image](https://user-images.githubusercontent.com/4809739/65375173-b6b28700-dc92-11e9-85c6-35e9965d2b16.png)

Notice how the TTF font uses subpixel rendering (=ClearType), while the OTF font does not. However, the OTF font seems better hinted than the TTF font... so it's not a clear win. I really wonder why the OTF font isn't using subpixel rendering, I don't know if it's caused by Windows, Qt, or me not using Qt correctly, or a combination of those.

The same result was obtained whether the fonts where installed in the system beforehand, or imported at runtime using QFontDatabase::addApplicationFont().

**Comment by dalboris** [comment]
Interestingly, when looking at the installed fonts in the Windows "Font Settings", neither the TTF font nor the OTF font uses subpixel rendering, but they still look quite different:

![image](https://user-images.githubusercontent.com/4809739/65375594-be285f00-dc97-11e9-8db5-9a698d1846eb.png)

Windows itself is quite inconsistent: none of the text in the UI of the settings themselves use subpixel rendering:

![image](https://user-images.githubusercontent.com/4809739/65375625-09db0880-dc98-11e9-841c-e1caeff146a6.png)

But many Windows apps do, like the File Explorer:

![image](https://user-images.githubusercontent.com/4809739/65375638-1fe8c900-dc98-11e9-8da9-6875e7305ae2.png)

The fact is that subpixel rendering is mostly useful for low-ppi LCD displays, less useful for high-ppi display, and even not applicable when the screen can be held in different orientation (portrait vs landscape). Things get even more complicated due to non-traditional subpixel matrix scheme such as AMOLED Pentile displays. Therefore, subpixel rendering of fonts is getting out of fashion lately, which is why Windows 10 doesn't use it anymore in some parts of the UI.

However, in the case of VGC, it is an app meant to be used on big LCD screens with relatively low ppi, and mostly oriented in the "natural" orientation of the screen (horizontal subpixels), so subpixel rendering makes sense for the VGC user interface.

For completeness, note that this is only true for the interface. Regarding the illustration/animation created with VGC, it shouldn't use subpixel rendering since there is no way to know what the output display will be once exported (e.g., mobile phone, printed, movie screen, etc.).

**Comment by dalboris** [comment]
For reference, here is the output when setting `printFontInfo = true;` in `main.cpp`:

When using OTF fonts:

```
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-Black.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-BlackIt.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-Bold.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-BoldIt.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-ExtraLight.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-ExtraLightIt.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-It.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-Light.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-LightIt.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-Medium.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-MediumIt.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-Regular.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-Semibold.otf
Added font file: widgets/fonts/SourceCodePro/OTF/SourceCodePro-SemiboldIt.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-Black.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-BlackIt.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-Bold.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-BoldIt.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-ExtraLight.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-ExtraLightIt.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-It.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-Light.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-LightIt.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-Regular.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-Semibold.otf
Added font file: widgets/fonts/SourceSansPro/OTF/SourceSansPro-SemiboldIt.otf
Font Family: Source Sans Pro
  Styles:
    Bold:
        weight:             75
        bold:               true
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Bold Italic:
        weight:             75
        bold:               true
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Italic:
        weight:             50
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Regular:
        weight:             50
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
Font Family: Source Code Pro
  Styles:
    Bold:
        weight:             75
        bold:               true
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Bold Italic:
        weight:             75
        bold:               true
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Italic:
        weight:             50
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Regular:
        weight:             50
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
```

When using TTF fonts:

```
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-Black.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-BlackIt.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-Bold.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-BoldIt.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-ExtraLight.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-ExtraLightIt.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-It.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-Light.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-LightIt.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-Medium.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-MediumIt.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-Regular.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-Semibold.ttf
Added font file: widgets/fonts/SourceCodePro/TTF/SourceCodePro-SemiboldIt.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-Black.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-BlackIt.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-Bold.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-BoldIt.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-ExtraLight.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-ExtraLightIt.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-It.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-Light.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-LightIt.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-Regular.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-Semibold.ttf
Added font file: widgets/fonts/SourceSansPro/TTF/SourceSansPro-SemiboldIt.ttf
Font Family: Source Sans Pro
  Styles:
    Black:
        weight:             87
        bold:               true
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Black Italic:
        weight:             87
        bold:               true
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Bold:
        weight:             75
        bold:               true
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Bold Italic:
        weight:             75
        bold:               true
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    ExtraLight:
        weight:             12
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    ExtraLight Italic:
        weight:             12
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Italic:
        weight:             50
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Light:
        weight:             25
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Light Italic:
        weight:             25
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Regular:
        weight:             50
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Semibold:
        weight:             63
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Semibold Italic:
        weight:             63
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       false
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
Font Family: Source Code Pro
  Styles:
    Black:
        weight:             87
        bold:               true
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Black Italic:
        weight:             87
        bold:               true
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Bold:
        weight:             75
        bold:               true
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Bold Italic:
        weight:             75
        bold:               true
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    ExtraLight:
        weight:             12
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    ExtraLight Italic:
        weight:             12
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Italic:
        weight:             50
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Light:
        weight:             25
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Light Italic:
        weight:             25
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Medium:
        weight:             57
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Medium Italic:
        weight:             57
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Regular:
        weight:             50
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Semibold:
        weight:             63
        bold:               false
        italic:             false
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
    Semibold Italic:
        weight:             63
        bold:               false
        italic:             true
        isBitmapScalable:   false
        isFixedPitch:       true
        isScalable:         true
        isSmoothlyScalable: true
        pointSizes:         [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
        smoothSizes:        [6, 7, 8, 9, 10, 11, 12, 14, 16, 18, 20, 22, 24, 26, 28, 36, 48, 72]
```

So it looks like that somehow, when using the OTF fonts, some font styles failed to be installed properly. All of this make it clear that on Windows, I really should be using the TTF fonts.

**Comment by dalboris** [comment]
After even more testing, it appears that the version of Source Sans Pro that I was using until now is the one hosted on Google Fonts (2.021-ro-1.076-it), and that it provides much better hinting that any of the versions provided on the Adobe GitHub repo (either 2.020-ro-1.075-it or 3.006, which look similar to me).

You can compare for yourself in the image below. On the left is the newer Adobe version, and on the right is the older Google version. The Google version on the right improves a lot of hinting problems, such as removing all the isolated pixels below the baseline, and providing more clarity.

![image](https://user-images.githubusercontent.com/4809739/65377398-8b886180-dcab-11e9-8d69-606256079b20.png)

Source Code Pro doesn't seem affected, both the Google version and Adobe version look properly hinted.

**Comment by dalboris** [comment]
Note: it is a shame that it isn't possible to specify whether we want hinting or not and how much in a qss stylesheet, white QFont proposed these options. Eventually, if I get rid of the QtWidgets dependency, I'll have much more flexibility in font rendering fine-tuning.

**Comment by dalboris** [comment]
Unfortunately, I also have problems with both OTF and TTF versions of Source Code Pro on macOS too.

Here is how it looks with 2.030-ro-1.050-it from Adobe GitHub repo (same results with either OTF and TTF fonts):

![image](https://user-images.githubusercontent.com/4809739/65379043-03fa1d00-dcc2-11e9-8e34-b5cce3d52b7e.png)

But here is how it looks if I use the 2.030-ro-1.050-it TTF version from Google Fonts:

![image](https://user-images.githubusercontent.com/4809739/65379927-e2ecf880-dcd0-11e9-9f63-95601e4df83f.png)

I'm not quite sure what to conclude here... except that the versions hosted at Google Fonts seem much more reliable than the one provided by Adobe. It is strange though that the OTF version of Source Sans Pro seems to work fine on macOS, it's just with Source Code Pro that I have problems.

**Comment by dalboris** [comment]
Alright, so the conclusion of all this was: both on Windows and macOS, it is better to use the TTF versions provided by Google Fonts, rather than any of the OTF or TTF versions provided in the Adobe GitHub repository. Therefore our current choice is very close to what we already had: we have just updated Source Code Pro to the latest version with italic variants, changed the font folder structure a little, and ship the OTF versions from Adobe too for potential future use (we don't use them yet on any platform).

**Comment by dalboris** [comment]
Coincidentally, while reading the list of bugs fixed in Qt 5.12.5, I found a bug specifically saying that Open Source Pro is rendered black on macOS 10.14:

https://bugreports.qt.io/browse/QTBUG-73097?filter=21443

So perhaps I can revisit the choice of using TTF fonts on macOS, and see if using 5.12.5 I can use the latest OTF fonts.

**Comment by dalboris** [comment]
I'm closing this since the TTFs fonts seem to work nicely anyway, so there's no reason to change something that's working for now, especially given all the other more important things to do.
